crates/astria-core/src/primitive/v1/asset.rs

```diff
@@ -83,6 +88,30 @@ impl Denom {
     pub fn to_base_denom(&self) -> Self {
         Self::from_base_denom(&self.base_denom)
     }
+
+    /// Create a new [`Denom`] with the given prefix removed.
+    ///
+    /// # Errors
+    ///
+    /// - if the denom does not have the given prefix.
+    pub fn remove_prefix(&self, prefix: &str) -> Result<Self, InvalidPrefixToRemove> {
+        let denom_trace = self
+            .to_string()
+            .strip_prefix(prefix)
+            .ok_or(InvalidPrefixToRemove {
```

Comment:
Use a lazy constructor to avoid allocation on every single call (i.e. even if you don't end up hitting the error case):
```suggestion
            .ok_or_or_else(|| InvalidPrefixToRemove {
```